core/primitives/src/shard_layout.rs

```diff
@@ -71,7 +74,17 @@ pub struct ShardLayoutV0 {
 /// Instead of using map, we just use a vec here because shard_id ranges from 0 to num_shards-1
 /// For example, if a shard layout with only shard 0 splits into shards 0, 1, 2, 3, the ShardsSplitMap
 /// will be `[[0, 1, 2, 3]]`
-type ShardSplitMap = Vec<Vec<ShardId>>;
+type ShardsSplitMap = Vec<Vec<ShardId>>;
```

Comment:
looks like ShardLayoutV1 is not a protocol struct, could we just use the new BTreeMap (ShardsParentMapV2) in all layout versions?

Reply:
Hmm, good question, I just treated it as immutable because it's versioned but you're right that it's not tagged as protocol schema. I think V1 can be considered a subset of V2 so it might just be ok to just implement the changes directly in V1. I'll have a look.

Reply:
Actually we need to be a bit more careful... `ShardLayout` is part of `GenesisConfig`, which is used to compute genesis json hash, which is probably used in some checks. If `ShardLayout` dramatically changes, genesis hash will change as well, I didn't consider that before.

However, as long as the first `ShardLayout`' serialisation is preserved, we should be fine... But we need to remove all these parameters from genesis at some point and just fix the hash.

Reply:
I'm going to leave it as is for now, it feels safer.